starters/next-react-query-tailwind/src/components/Greeting/Greeting.tsx

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from "react-query";
+
+export function Greeting() {
+  const {
+    data: message,
+    isLoading,
+    error,
+  } = useQuery<string>('hello', async () => {
+    return await fetch('https://api.starter.dev/hello?greeting=from This Dot Labs!').then((res) =>
+      res.text()
+    );
```

Comment:
Personally, I think this is a bit strange to read since it's mixing Promises and async/await patterns. I think it would be best if this was rewritten to be:

```suggestion
    const resp = await fetch('https://api.starter.dev/hello?greeting=from This Dot Labs!');
    const text = await resp.text();
    return text;
```

This way it's all using the async/await pattern. We arguably should also add a `try...catch` around it as a best practice.